File: pronic.scala
def pronic(i: Int, j: Int): List[Int] = {
    if (i < 0 || j < 0) throw new Exception("All values must be greater than zero.")
    def pronicR(s: Int, e: Int, res: List[Int]): List[Int] = {
        if (e < s) res
        else pronicR(s, e - 1, (e * (e - 1)) :: res)
    }
    pronicR(i, j, Nil)
}

File: Pronic.java
class Pronic {
	public static List<BigInteger> pronic(final Integer i, final Integer j) {
		if (i < 0 || j < 0) return new ArrayList<>(Arrays.asList(BigInteger.ZERO));
		class PronicR {
			private List<BigInteger> pronicR(final Integer start, final Integer end, final List<BigInteger> result) {
				if (end < start) return result;
				else
					result.add(0, BigInteger.valueOf((end * (end - 1))));
					return pronicR(start, (end - 1), result);
			}
		}
		return new PronicR().pronicR(i, j, new ArrayList<BigInteger>());
	}
}
